lib/active_model/serializer/associations.rb

```diff
@@ -92,7 +92,8 @@ def associations(include_tree = DEFAULT_INCLUDE_TREE)
 
         Enumerator.new do |y|
           self.class._reflections.each do |reflection|
-            next unless include_tree.key?(reflection.name)
+            name = reflection.options.fetch(:key, reflection.name)
```

Comment:
lgtm, though we need to be careful with this include_tree code. It's introduced some bugs and complexity